src/sentry/web/frontend/setup_wizard.py

```diff
@@ -60,85 +63,122 @@ def get(self, request: HttpRequest, wizard_hash) -> HttpResponse:
-                keys_map[key.project_id] = [key]
-            else:
-                keys_map[key.project_id].append(key)
+        ).order_by("-date_created")
+
+        # TODO: Make wizard compatible with hybrid cloud. For now, we combine all region data for these
+        # responses, but project names/slugs aren't unique across regions which could confuse some users.
+        # Wizard should display region beside project/orgs or have a step to ask which region.
+
+        region_data_map = defaultdict(lambda: defaultdict(list))
+        # {'us': {'org_ids': [...], 'projects': [...], 'keys': [...]}}
+
+        for mapping in org_mappings:
+            region_data_map[mapping.region_name]["org_ids"].append(mapping.organization_id)
+
+        for region_name, region_data in region_data_map.items():
+            org_ids = region_data["org_ids"]
+            region_data["projects"] = project_service.get_many_by_organizations(
+                region_name=region_name, organization_ids=org_ids
+            )
+
+        for region_name, region_data in region_data_map.items():
+            project_ids = [rpc_project.id for rpc_project in region_data["projects"]]
+            region_data["keys"] = project_key_service.get_project_keys_by_region(
+                region_name=region_name,
+                project_ids=project_ids,
+                role=ProjectKeyRole.store,
+            )
+
+        org_mappings_map = {}
+        for mapping in org_mappings:
+            status = OrganizationStatus(mapping.status)
+            serialized_mapping = {
+                "id": mapping.organization_id,
+                "name": mapping.name,
+                "slug": mapping.slug,
+                "region": mapping.region_name,
+                "status": {"id": status.name.lower(), "name": status.label},
+            }
+            org_mappings_map[mapping.organization_id] = serialized_mapping
+
+        keys_map = defaultdict(list)
+        for key in region_data["keys"]:
+            serialized_key = {
+                "dsn": {"public": key.dsn_public},
+                "isActive": key.is_active,
+            }
+            keys_map[key.project_id].append(serialized_key)
 
         filled_projects = []
 
-        for project in projects:
+        for project in region_data["projects"]:
+            # TODO(Leander) check if this is safe to replace with rpc project
             enriched_project = serialize(project)
-            enriched_project["organization"] = serialize(orgs_map[project.organization_id])
-            enriched_project["keys"] = serialize(keys_map.get(project.id, []))
+            enriched_project = {
+                "slug": project.slug,
+                "id": project.id,
+                "status": STATUS_LABELS.get(project.status, "unknown"),
+            }
+            # The wizard only reads the `slug` field so serializing the mapping should work fine
+            # Maybe security concerns though?
```

Comment:
What concerns do you have here? It looks like we would be exposing less data than before.

Reply:
Oh that was while i was developing, i was serializing the whole org mapping including the customer ID from Stripe which is what I was talking about here 😅 

i'll remove the comment :+1: